pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -455,21 +454,16 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, workloadLabels map[string]st
 			//   case 2: the workload xDS connection has not been established,
 			//           also no need to trigger a full push here.
 			if connection.modelNode.IPAddresses[0] == id {
-				fullPush = true
+				// There is a possibility that the pod comes up later than endpoint.
+				// So no endpoints add/update events after this, we should request
+				// full push immediately to speed up sidecar startup.
+				_ = connection.modelNode.SetServiceInstances(s.Env)
```

Comment:
I think this will cause a data race if another push is active, we don't protect the proxy with a mutex?

Reply:
yes, there is a race. But doesnot matter much, as this will trigger a full push afterwards

Reply:
It just might trigger the `-race` test. but it doesn't seem to so maybe its ok

Reply:
Actually why do we need to set this here? It will be set in pushConnection anyways?